repos/system_upgrade/el7toel8/actors/selinux/selinuxapplycustom/actor.py

```diff
@@ -0,0 +1,102 @@
+import os
+from shutil import rmtree
+
+from leapp.actors import Actor
+from leapp.models import SELinuxModules, SELinuxCustom, SELinuxRequestRPMs
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+from leapp.libraries.stdlib import run, CalledProcessError
+
+
+WORKING_DIRECTORY = '/tmp/selinux/'
+
+
+class SELinuxApplyCustom(Actor):
+    '''
+    Re-apply SELinux customizations from RHEL-7 installation
+
+    Re-apply SELinux policy customizations (custom policy modules and changes
+    introduced by semanage). Any changes (due to incompatiblity with RHEL-8
+    SELinux policy) are reported to user.
+    '''
+    name = 'selinuxapplycustom'
+    consumes = (SELinuxCustom, SELinuxModules)
+    produces = ()
+    tags = (ApplicationsPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        # cil module files need to be extracted to disk in order to be installed
+
+        # clear working directory
+        rmtree(WORKING_DIRECTORY, ignore_errors=True)
+
+        try:
+            os.mkdir(WORKING_DIRECTORY)
+        except OSError:
+            self.log.warning("Failed to create working directory! Aborting.")
+            return
+
+        # import custom SElinux modules
+        for semodules in self.consume(SELinuxModules):
+            self.log.info("Processing custom SELinux policy modules. Count: %d.", len(semodules.modules))
+            for module in semodules.modules:
+                cil_filename = os.path.join(WORKING_DIRECTORY, "%s.cil".format(module.name))
+                self.log.info("Installing module %s on priority %d.", module.name, module.priority)
+                if module.removed:
+                    self.log.warning("The following lines where removed because of incompatibility: \n%s",
+                                     '\n'.join(module.removed))
+                # write module content to disk
+                try:
+                    with open(cil_filename, 'w') as cil_file:
+                        cil_file.write(module.content)
+                except OSError as e:
+                    self.log.warning("Error writing %s : %s", cil_filename, str(e.stderr))
```

Comment:
Does OSError have stderr at all? It's on many places.
```
>>> aaa = OSError('sfsfs')
>>> aaa.stder
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'exceptions.OSError' object has no attribute 'stderr'
```

Reply:
Only our CalledProcessError does have it

OSError has errno however